### PR TITLE
update table in assessments

### PR DESCRIPTION
**Pull request by yash6102000**
integrate table and pagination in assessments

### CODE REVIEW

File: app/components/tests/TestList.tsx [hunk 1]
```diff
@@ -52,7 +70,124 @@ const TestList = ({
...
+        <InviteCandidatePopup
+          openInvitePopup={candidatePopupOpen}
+          setOpenInvitePopup={setCandidatePopupOpen}
+          testName={data.name}
+          testId={data.id}
+        />
+      </>
+    )
+  }
+  const testsColumn = [
```
Comment (richa-copods) [inline app/components/tests/TestList.tsx:167]: Give proper type to `testsColumn`. Better if you rename(testColumns) it too.

File: app/interface/Interface.tsx [hunk 2]
```diff
@@ -314,7 +314,7 @@ export interface SectionWiseResults {
 export interface column {
   title: string
   field: string
-  render?: (data: any) => JSX.Element
+  render?: (data: any, index: number) => JSX.Element
```
Comment (richa-copods) [inline app/interface/Interface.tsx:317]: Give type to `data`

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/tests/TestList.tsx (hunk 1 above)
Comment (richa-copods) [reply app/components/tests/TestList.tsx]: Types are not added, please add them

File: app/components/tests/TestList.tsx [hunk 3]
```diff
@@ -52,7 +70,124 @@ const TestList = ({
...
+  const [testsCurrentPage, setTestsCurrentPage] = useState(
+    testLoaderData.testsCurrentPage
+  )
+  const [testsPageSize, setTestsPageSize] = useState(5)
+  const [candidatePopupOpen, setCandidatePopupOpen] = useState<boolean>(false)
+  const [showDeletePopup, setShowDeletePopup] = useState(false)
+  const [deleted, setDeleted] = useState(false)
+
+  useEffect(() => {
+    if (deleted === true) {
```
Comment (richa-copods) [inline app/components/tests/TestList.tsx:82]: It can be written as `if(deleted)`

File: app/components/tests/TestList.tsx [hunk 4]
```diff
@@ -52,7 +70,124 @@ const TestList = ({
...
+        <InviteCandidatePopup
+          openInvitePopup={candidatePopupOpen}
+          setOpenInvitePopup={setCandidatePopupOpen}
+          testName={data.name}
+          testId={data.id}
+        />
+      </>
+    )
+  }
+  const testColumns = [
```
Comment (richa-copods) [inline app/components/tests/TestList.tsx:167]: Add types for Column

File: app/components/tests/TestList.tsx [hunk 5]
```diff
@@ -52,7 +70,124 @@ const TestList = ({
...
+      title: 'Created By',
+      field: 'createdAt',
+      render: CreatedByDataCell,
+      width: '20%',
+    },
+    { title: 'Action', field: 'action', render: TestInvite, width: '10%' },
+  ]
+  useEffect(() => {
+    navigate(
+      `?index=&data=%7B"orderBy"%3A%7B"createdAt"%3A"desc"%7D%7D&TestPage=${testsCurrentPage}&TestItems=${testsPageSize}`
```
Comment (richa-copods) [inline app/components/tests/TestList.tsx:187]: Use proper naming conventions for query parameters, `index`, `data` start with small case whereas `TestPage` and `TestItems` start with capital letters

File: app/models/tests.server.ts [hunk 6]
```diff
@@ -16,15 +16,28 @@ export async function getTestById({ id }: Pick<Section, 'id'>) {
...
+    },
+  })
+  return testCount
+}
 export async function getAllTests(
   filterData: { orderBy: { name?: string; createdAt?: string } },
-  workspaceId: string
+  workspaceId: string,
+  testsItemsPerPage = 1,
+  testsCurrentPage = 5
```
Comment (richa-copods) [inline app/models/tests.server.ts:32]: Why currentPage is 5?

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/models/tests.server.ts (hunk 6 above)
Comment (yash6102000) [reply app/models/tests.server.ts:32]: it is a fallback value

File: app/interface/Interface.tsx (hunk 2 above)
Comment (akshay-copods) [reply app/interface/Interface.tsx:317]: In interface we are keeping render function parameter `data` type as `any` for now. In future we will update it with proper type.

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**